lib/Partitioner/Partitioner.cpp

```diff
@@ -1005,9 +1028,30 @@ llvm::Error Partitioner::Partition(CompilationContext &cctx) {
   for (auto i = funcToBackend.begin(); i != funcToBackend.end(); ++i) {
     module_->eraseFunction(i->first);
   }
+
   auto funcList = module_->getFunctions();
+  if (logPartition) {
+    LOG(INFO) << "The number of partitions is : " << funcList.size()
+              << ", and the DAG is dumped into DAG.dot file.\n";
+    dumpDAG("DAG.dot");
+  }
+  int i = 0;
   for (Function *subF : funcList) {
     (void)subF;
+    if (logPartition) {
+      LOG(INFO) << "\t Partition " << i << ":\n"
+                << "\t\t Name :\t" << subF->getName().str() << "\n"
+                << "\t\t BackendKind :\t"
+                << mapping.getPartitionBackendName(subF) << "\n"
+                << "\t\t Memory :\t"
+                << mapping.getGraphMemInfo(subF).getTotalMemSize() << "\n"
+                << "\t\t LogicalDeviceIDs :\t"
+                << mapping.getLogicalDeviceIDList(subF)[0] << "\n";
+    }
+    if (dumpPartition) {
+      subF->dumpDAG("part_" + std::to_string(i) + ".dot");
```

Comment:
Based on my debugging experience, I'd suggest including the backend name into the file name used for dumping the DAG. E.g. something like `part_2_CPU.dot`. It is also helpful if the name of `subF` (and may be other infos printed a couple of lines above) would be somehow present in the dumped file and rendered by the dot files viewer, so that it is easier to correlate this file to the function.

Reply:
Now the name is made of "part_"+number+function_name+backendName. E.g:
part_0_en2gr_part1_CPU.dot

Reply:
> part_0_en2gr_part1_CPU.dot

 It looks a bit inconsistent: `part_0` with an underscore in the beginning, but `part1` without underscores at the end. Let's remove the first underscore as well then. Having `part0` and `part1` in the same name is also a bit odd. Maybe use `partition0` for the initial part?

Reply:
The name is updated: partitionLogicalID + "__" + functionName + "__" + BackendName.
e.g.

```
wangm-mbp:build_Debug1 wangm$ ./tests/PartitionerTest --gtest_filter=*PartitionerTest.logicalIDTest0* -dump-partition
Note: Google Test filter = *PartitionerTest.logicalIDTest0*
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from PartitionerTest
[ RUN      ] PartitionerTest.logicalIDTest0
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0618 13:01:14.920496 342365632 Partitioner.cpp:1034] The number of partitions is : 3, and the DAG is dumped into DAG.dot file.
I0618 13:01:14.921118 342365632 Partitioner.cpp:73] Writing dotty graph for DAG after graph partitioning: DAG.dot
I0618 13:01:14.921526 342365632 Partitioner.cpp:1043] 	 Partition 0:
		 Name :	main_part1
		 BackendKind :	Interpreter
		 Memory :	848
		 LogicalDeviceIDs :	0
Writing dotty graph for Function to: partitionLogicalID_0__main_part1__Interpreter.dot
I0618 13:01:14.922056 342365632 Partitioner.cpp:1043] 	 Partition 1:
		 Name :	main_part2
		 BackendKind :	Interpreter
		 Memory :	1496
		 LogicalDeviceIDs :	1
Writing dotty graph for Function to: partitionLogicalID_1__main_part2__Interpreter.dot
I0618 13:01:14.922384 342365632 Partitioner.cpp:1043] 	 Partition 2:
		 Name :	main_part3
		 BackendKind :	Interpreter
		 Memory :	608
		 LogicalDeviceIDs :	0
Writing dotty graph for Function to: partitionLogicalID_0__main_part3__Interpreter.dot
[       OK ] PartitionerTest.logicalIDTest0 (3 ms)
[----------] 1 test from PartitionerTest (3 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (3 ms total)
[  PASSED  ] 1 test.
```